sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -431,120 +521,47 @@ protected EventProcessorClient()
-                            {
-                                throw new InvalidOperationException(string.Format(CultureInfo.CurrentCulture, Resources.CannotStartEventProcessorWithoutHandler, nameof(ProcessExceptionAsync)));
-                            }
-
-                            // We expect the token source to be null, but we are playing safe.
-
-                            RunningTaskTokenSource?.Cancel();
-                            RunningTaskTokenSource = new CancellationTokenSource();
-
-                            // Initialize our empty ownership dictionary.
-
-                            InstanceOwnership = new Dictionary<string, PartitionOwnership>();
-
-                            // Start the main running task.  It is responsible for managing the partition pumps and for partition
-                            // load balancing among multiple event processor instances.
-
-                            ActiveLoadBalancingTask = RunAsync(RunningTaskTokenSource.Token);
-                        }
-                    }
-                }
-                finally
-                {
-                    RunningTaskSemaphore.Release();
-                }
-            }
-        }
+        protected override PartitionOwnership CreatePartitionOwnership(string partitionId,
+                                                                       long? offset,
+                                                                       long? sequenceNumber,
+                                                                       DateTimeOffset? lastModifiedTime,
+                                                                       string eTag) => new PartitionOwnership(FullyQualifiedNamespace, EventHubName, ConsumerGroup, Identifier, partitionId, offset, sequenceNumber, lastModifiedTime, eTag);
 
         /// <summary>
-        ///   Stops the event processor.  In case it isn't running, nothing happens.
+        ///   Creates an <see cref="EventHubConnection" /> instance.  The returned instance must not be returned again by other
+        ///   <see cref="CreateConnection" /> calls.
         /// </summary>
         ///
-        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        /// <returns>A new <see cref="EventHubConnection" /> instance.</returns>
         ///
-        public virtual async Task StopAsync()
-        {
-            if (ActiveLoadBalancingTask != null)
-            {
-                await RunningTaskSemaphore.WaitAsync().ConfigureAwait(false);
+        /// <remarks>
+        ///   The abstract <see cref="EventProcessorBase{T}" /> class has ownership of the returned connection and, therefore, is
+        ///   responsible for closing it.  Attempting to close the connection in the derived class may result in undefined behavior.
+        /// </remarks>
+        ///
+        protected override EventHubConnection CreateConnection() => ConnectionFactory();
```

Comment:
These seem to have worked out alright.

Reply:
Please keep a list of your ideas for revising the base/concrete relationship.   We'll huddle up after we get things release-ready and see if we can revise the design.